test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java

```diff
@@ -2139,9 +2125,9 @@ public void handleResponse(TransportResponse response) {
             @Override
             public void handleException(TransportException exp) {
                 try {
-                    assertTrue(threadPool.getThreadContext().getResponseHeaders().containsKey("foo.bar"));
-                    assertEquals(1, threadPool.getThreadContext().getResponseHeaders().get("foo.bar").size());
-                    assertEquals("baz", threadPool.getThreadContext().getResponseHeaders().get("foo.bar").get(0));
+                    assertThat(threadPool.getThreadContext().getResponseHeaders(), hasKey("foo.bar"));
```

Comment:
I don't have a strong opinion about hamcrest for equals checks, but for things like hasKey here, it gives far superior failure information compared to a simple assertTrue/False.